### ISSUE TITLE
Running `eslint .` doesn't check all files

### ISSUE THREAD

**Issue by jamesob** [eslint/eslint#7296]
**Tell us about your environment**

* **ESLint Version:** 3.6.1
* **Node Version:** 6.7.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

```js
{
    "parser": "babel-eslint",
    "env": {
        "browser": true,
        "commonjs": true,
        "node": true,
    },
    "settings": {
        "ecmascript": 6,
        "jsx": true,
    },
    "plugins": [
        "react",
    ],
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended",
    ],
    "rules": {
        "max-len": ["error", 80, 2],

        // Default to single quotes and raise an error if something
        // else is used
        "quotes": [2, "single", { "allowTemplateLiterals": true }],

        // Don't warn us about using console.log (though that should only
        // happen in the dev environment).
        "no-console": [0],
    },
    "globals": {
        "__DEV__": false,
    },
}

```

**What did you do? Please include the actual source code causing the issue.**

```sh
 $ eslint design

$ eslint design/components/base/index.jsx


/home/james/tempo/tempocom/frontend/design/components/base/index.jsx
   19:50  error  'sm' is missing in props validation          react/prop-types
   22:43  error  'sm' is missing in props validation          react/prop-types
   23:21  error  'children' is missing in props validation    react/prop-types
   48:32  error  'subheader' is missing in props validation   react/prop-types
   64:27  error  'children' is missing in props validation    react/prop-types
   66:26  error  'subheader' is missing in props validation   react/prop-types
   68:31  error  'subheader' is missing in props validation   react/prop-types
  101:41  error  'accept' is missing in props validation      react/prop-types
  102:55  error  'style' is missing in props validation       react/prop-types
  103:25  error  'children' is missing in props validation    react/prop-types
  127:21  error  'sidebar' is missing in props validation     react/prop-types
  129:23  error  'children' is missing in props validation    react/prop-types
  153:25  error  'children' is missing in props validation    react/prop-types
  176:23  error  'header' is missing in props validation      react/prop-types
  178:21  error  'children' is missing in props validation    react/prop-types
  196:18  error  'isActive' is missing in props validation    react/prop-types
  213:68  error  'status' is missing in props validation      react/prop-types
  221:25  error  'isActive' is missing in props validation    react/prop-types
  235:20  error  'status' is missing in props validation      react/prop-types
  237:25  error  'status' is missing in props validation      react/prop-types
  239:25  error  'status' is missing in props validation      react/prop-types
  241:25  error  'status' is missing in props validation      react/prop-types
  254:48  error  'titleStyle' is missing in props validation  react/prop-types
  255:23  error  'title' is missing in props validation       react/prop-types
  258:23  error  'children' is missing in props validation    react/prop-types
  263:20  error  'linkTo' is missing in props validation      react/prop-types
  265:63  error  'linkTo' is missing in props validation      react/prop-types
  282:36  error  'children' is missing in props validation    react/prop-types
  291:18  error  'width' is missing in props validation       react/prop-types
  298:14  error  'style' is missing in props validation       react/prop-types
  300:39  error  'children' is missing in props validation    react/prop-types

✖ 31 problems (31 errors, 0 warnings)

```

**What did you expect to happen?**

I expected ESLint to check the design/ directory (including all files and subdirectories).

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint apparently didn't check all of the files in the directory.

**Comment by platinumazure** [comment]
When you run ESLint with a folder path, ESLint looks for files based on extensions you specify. By default, that's just `.js`. Since you haven't specified any other extensions, the `.jsx` files aren't linted.

You have two options:

1. Use `--ext`: `eslint design --ext js,jsx`
    a. Or just `eslint design --ext jsx`
1. Use a glob pattern instead of a folder path: `eslint design/**/*.jsx`

**Comment by jamesob** [comment]
Ah, okay, thanks for the explanation. Might be worth some kind of warning to stderr if there are no files processed and a number of potentially compatible files (i.e. files ending in '.jsx') present.

**Comment by platinumazure** [comment]
I think I could get behind your suggestion- it's the inverse of the warning we emit when a file is explicitly specified but ignored to `.eslintignore`. Instead of warning a user that a specific file was ignored, we could say that a specified path or glob didn't yield any files for the linter  to lint.

Let's see what the rest of the team thinks about this.

**Comment by nzakas** [comment]
@jamesob if you'd like to propose something, please flesh out the idea and explain how you think it should work and under what circumstances. What you're described isn't specific enough for us to be able to evaluate.

**Comment by jamesob** [comment]
@nzakas I don't have the time/interest to do that. Thanks for the tool!

**Comment by platinumazure** [comment]
For what it's worth, I like the idea enough that I'll create a new issue with a proposal at some point. I'll try to remember to link to that issue from here when I create it, but I make no promises.
